### DISCUSSION TITLE
Chaotic websites

### DISCUSSION THREAD

**Discussion by modir** [laminas/technical-steering-committee#172] [Q&A]
We have at the moment at least two "official" websites about laminas. One is https://laminas.dev/ and the other one is https://getlaminas.org/. Who is responsible for them? What is the purpose of each? Why two different websites?

Furthermore do we really need the blog at https://getlaminas.org/blog/? Or could this be integrated into discourse?

**Comment by froschdesign** [comment]
@modir 
> Or could this be integrated into discourse?

To be clear: Discourse is not the central platform of this project.

@weierophinney 
We had already discussed this topic in the past. The problem is in each `composer.json`:

```json
"homepage": "https://laminas.dev",
```

But the redirect didn't work: https://github.com/laminas/laminas.dev/pull/15

**Reply by weierophinney** [reply]
It may be best for us to change the "homepage" metadata for each package, then.

The rationale for having laminas.dev and getlaminas.org is water under the bridge at this point. We use laminas.dev for documentation purposes (docs.laminas.dev) and for some project integrations (it provides some webhooks used within the project), but getlaminas.org has been the primary project website since we launched the project fully in 2020. The fact that `laminas.dev` is used in the packages is a detail from when we were rewriting the packages from ZF.

We can script this change across the repos, applying it to the most recent release branch; I know @michalbundyra has some scripts that can make this relatively quick.

**Comment by modir** [comment]
Followup question: Is there a repo for the getlaminas.org website where I can commit changes? E.g it would be great to have a section "learning" where we link to books and tutorials and other learning material provided by the community. Or how do you want to get feedback about what to change on the website?

**Reply by froschdesign** [reply]
> Is there a repo for the getlaminas.org website…

Sure. The repository has the same name as the website: https://github.com/laminas/getlaminas.org

> E.g it would be great to have a section "learning" where we link to books and tutorials and other learning material provided by the community. 

Do you have concrete suggestions?

**Reply by froschdesign** [reply]
A "Learn" section is definitely needed and I would love to have it. But the focus should be on Mezzio, because a new application should definitely be created with Mezzio. The benefits are:

- the entry is easier
- the learning curve is flatter
- is much more open, as any PSR middleware can be used
- provides command line tooling
- can run with application servers like Roadrunner or Swoole
- Tests are easier to realise

**Reply by modir** [reply]
Just because one is better doesn't mean we shouldn't honor the work of the others. From my point of view we should list all of them. We can still write at the top that Mezzio is now preferred. But there are still many companies with the normal MVC and they hire new employees as well which need training. So there is still valid reasons to keep this information available as well.

**Reply by froschdesign** [reply]
> …doesn't mean we shouldn't honor the work of the others.

That's not the topic here.

> So there is still valid reasons to keep this information available as well.

Right, but here we can also use the documentation, for example:

- https://docs.laminas.dev/mvc/
- https://docs.laminas.dev/laminas-mvc/

**Reply by froschdesign** [reply]
I mean something like this: https://docs.astro.build/en/community-resources/content/

**Comment by modir** [comment]
And who can write blog posts? Are blog articles by the community of interest to the project?

**Comment by modir** [comment]
> Do you have concrete suggestions?

Yes. I have:
These two books:
https://www.packtpub.com/product/php-web-development-with-laminas/9781803245362
https://www.amazon.de/Laminas-MVC-Framework-Adam-Omelak/dp/B088B24K5S
Once I finished the changes you recommended to my rewrite of Oleg's book we could link it here too.

Then this video:
https://www.youtube.com/watch?v=mvuOErwUcPY

Or this starter kit:
http://laminasstarterkit.omelak.com/tutorials/9pXaLWv7gPU

**Reply by modir** [reply]
Or this tutorial:
https://www.vistomedia.com/laminas-user-authentication-using-lmcuser/

Or these articles:
https://matthewsetter.com/tags/laminas/

**Reply by froschdesign** [reply]
"Mezzio: Getting Started"
by Matthew Setter
https://www.pluralsight.com/courses/mezzio-getting-started

**Reply by modir** [reply]
I see there is more and more coming together. In my point of view we should list them on the website. As well to show that we honor the time people spent on writing about the framework. They are making marketing for the framework which is very valuable.

**Reply by froschdesign** [reply]
"Mezzio Essentials" is another book from @settermjd: https://matthewsetter.com/books-and-courses/

**Reply by boesing** [reply]
> "Mezzio Essentials" is another book from @settermjd: https://matthewsetter.com/books-and-courses/

I had a quick glance into this a few years back and that was pretty good!
